### PR TITLE
Add common tags for stats backends that support it

### PR DESCRIPTION
**Pull request by narcsfz**
## Description

Opentsdb and statsd both support common tags that can be attached to all the metrics we emitted. They are useful to identify the latency at keyspace or shard level. However there is no easy way to integrate with that. 

This PR does 2 things:
1. passes the common tags from a comma-separated string and we can tag the stats client directly
2. minor change on opentsdb and statsd init process by adding `InitWithoutServenv` so that we can init them without servenv. This can be useful to reuse the stats backend on other binaries that does not use servenv (e.g., vtorc, vtbench)

## Related Issue(s)

- https://github.com/vitessio/vitess/issues/7647

## Checklist
- [x] Should this PR be backported?
- [x] Tests were added or are not required
- [ ] Documentation was added or is not required

## Impacted Areas in Vitess
Components that this PR will affect:

- [x]  Query Serving
- [ ]  VReplication
- [ ]  Cluster Management
- [ ]  Build/CI
- [ ]  VTAdmin

### CODE REVIEW

**Comment by narcsfz** [comment]
@deepthi sorry for the ping - would you mind taking a look when you get a chance? Thanks! :-)

**Comment by narcsfz** [comment]
Gentle reminder :-)

**Review by deepthi — COMMENTED** [review]
Looks fine to me.
One question: why did you decide to extract out `InitWithoutServenv`? I don't see it being called outside `servenv.OnRun`.

**Comment by narcsfz** [comment]
> Looks fine to me.
> One question: why did you decide to extract out `InitWithoutServenv`? I don't see it being called outside `servenv.OnRun`.

I extract it out so that it is possible to reuse stats in other Vitess components that does not set up `servenv` (e.g., vtorc, vtbench, etc.). For example, in the plugin we can have `statsd.InitWithoutServenv("vtbench")` to support statsd in vtbench

**Review by deepthi — APPROVED** [review]

File: go/stats/export.go [hunk 1]
```diff
@@ -46,6 +46,9 @@ var statsBackend = flag.String("stats_backend", "", "The name of the registered
 var combineDimensions = flag.String("stats_combine_dimensions", "", `List of dimensions to be combined into a single "all" value in exported stats vars`)
 var dropVariables = flag.String("stats_drop_variables", "", `Variables to be dropped from the list of exported variables.`)
 
+// CommonTags is a comma-separated list of common tags for stats backends
+var CommonTags = flag.String("stats_common_tags", "", `Comma-separated list of common tags for the stats backend. It provides both label and values. Example: label1:value1,label2:value2`)
```
Comment (dweitzman) [inline go/stats/export.go:50]: Seems like this should use `flagutil.StringMapValue`

**Pull request merged**